geode-docs/managing/security/method_invocation_authorizers.html.md.erb

```diff
@@ -0,0 +1,165 @@
+
+A fully flexible `MethodInvocationAuthorizer` that allows methods to be invoked during the query execution only if the the following conditions are met:
+
+- The method is not considered permanently forbidden by the [RestrictedMethodAuthorizer](method_invocation_authorizers.html#restrictedMethodAuthorizer).
+- The method does not belong to <%=vars.product_name%>, or does belong but it's considered safe by the [RestrictedMethodAuthorizer](method_invocation_authorizers.html#restrictedMethodAuthorizer).
+- The fully qualified method name matches at least one of the pre-configured regular expressions.
+
+When correctly configured, this authorizer implementation addresses the four main security risks: `Java Reflection`, `Cache Modification`, `Region Modification` and `Region Entry Modification`.   
+For the statement to remain true, however, the regular expressions used must be exhaustively studied and configured so no mutator methods ever match. If the regular expressions are not restrictive enough, 
+the `Region Entry Modification` security risk might be potentially re-introduced: users with the `DATA:READ:RegionName` privilege will be able to execute methods (even those modifying the entry) on 
+the objects stored within the region and on instances used as bind parameters of the query. 
+
+**Note:**
+This authorizer must be used with extreme care, it is the most powerful in terms of flexibility and versatility (full control through regular expressions regarding what to allow and what to forbid); 
+but it's also the most dangerous as one small mistake in the configured regular expressions can wronly allow a wide variety of non safe methods to be executed.
+
+**Note:**
+Usage of this authorizer implementation is only recommended for scenarios on which the user knows exactly what code is deployed to the cluster (how and when), allowing a correct configuration of the regular expressions used.
+It might also be used on clusters on which all entries stored are immutable.
+
+## <a id="implementing_custom_authorizer"></a>Custom Authorizers
+
+### How Authorization Works
+
+It's important to note that the query engine does not have any information about the actual type of the objects while pre-processing or parsing the query itself, neither can it obtain these details before actually executing the query.
+The actual check to determine whether a method is allowed or not must be executed while the objects are being traversed by the query engine in runtime.
+
+The query engine, however, remembers whether a specific method has been already authorized or not for the current query execution context, meaning that **the authorization will be executed only once in the lifetime of a particular query 
+for every new method seen while traversing the objects**. Nevertheless, the authorizer implementation must be highly performant as it will be invoked by <%=vars.product_name%> in runtime during the actual query execution.
+
+### Implement Method Authorizer 
```

Comment:
Maybe this should be "How to Implement a Method Authorizer" or "Implementing a Method Authorizer"

Reply:
👍